erts/emulator/beam/erl_utils.h

```diff
@@ -231,6 +232,17 @@ ERTS_GLB_INLINE Sint erts_cmp(Eterm a, Eterm b, int exact, int eq_only) {
     return erts_cmp_compound(a,b,exact,eq_only);
 }
 
+/*
+ * Only to be used for the *internal* sort order of flatmap keys.
+ */
+ERTS_GLB_INLINE Sint erts_cmp_flatmap_keys(Eterm key_a, Eterm key_b) {
+    if (is_atom(key_a) && is_atom(key_b)) {
```

Comment:
I was wondering - could similar be done for all immediate values, not just atoms?

Reply:
I think that's already what `erts_cmp` does, no? Atoms were the only exception for intermediates.

Reply:
It could be done for all immediates, but then the algorithm for comparing two maps would be even more expensive. It would probably have to make temporary sorted copies of the key arrays before comparing them.

Reply:
@sverker but that would only be for `<` and `>` comparison, right? `==` should still be fast because the keys have to match anyway?

Reply:
Correct.

Reply:
I see. I do think atoms are the biggest win given record-like usage patterns, so it makes sense to do this for atoms only, if it helps keep the implementation simpler